src/main/java/gregtech/api/pattern/TraceabilityPredicate.java

```diff
@@ -95,7 +96,7 @@ public TraceabilityPredicate sort() {
      * Add tooltips for candidates. They are shown in JEI Pages.
      */
     public TraceabilityPredicate addTooltips(String... tips) {
-        if (tips.length > 0) {
+        if (FMLCommonHandler.instance().getSide() == Side.CLIENT && tips.length > 0) {
```

Comment:
This doesn't fix the turbines because the i18n call is executed in the parameter list (before we get to this function) in the large turbine's `createStructurePattern()`.. AFAICT there's no i18N usage here that needs to be saved though might as well not make the server execute it.

To fix the turbine, need to add something like this:

```diff
diff --git a/src/main/java/gregtech/common/metatileentities/multi/electric/generator/MetaTileEntityLargeTurbine.java b/src/main/java/gregtech/common/metatileentities/multi/electric/generator/MetaTileEntityLargeTurbine.java
index 9369b42e9..864368311 100644
--- a/src/main/java/gregtech/common/metatileentities/multi/electric/generator/MetaTileEntityLargeTurbine.java
+++ b/src/main/java/gregtech/common/metatileentities/multi/electric/generator/MetaTileEntityLargeTurbine.java
@@ -27,6 +27,8 @@ import net.minecraft.util.text.TextFormatting;
 import net.minecraft.world.World;
 import net.minecraftforge.fluids.FluidStack;
 import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+import net.minecraftforge.fml.relauncher.Side;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
@@ -142,6 +144,11 @@ public class MetaTileEntityLargeTurbine extends FuelMultiblockController impleme
 
     @Override
     protected BlockPattern createStructurePattern() {
+        String limitedMsg = "gregtech.multiblock.pattern.error.limited.1";
+        // only attempt i18n on client side
+        if (FMLCommonHandler.instance().getSide() == Side.CLIENT) {
+            limitedMsg = I18n.format(limitedMsg, GTValues.VN[tier]);
+        }
         return FactoryBlockPattern.start()
                 .aisle("CCCC", "CHHC", "CCCC")
                 .aisle("CHHC", "RGGR", "CHHC")
@@ -153,7 +160,7 @@ public class MetaTileEntityLargeTurbine extends FuelMultiblockController impleme
                         .filter(mte -> (mte instanceof ITieredMetaTileEntity) && (((ITieredMetaTileEntity) mte).getTier() >= tier))
                         .toArray(MetaTileEntity[]::new))
                         .addTooltips("gregtech.multiblock.pattern.clear_amount_3")
-                        .addTooltips(I18n.format("gregtech.multiblock.pattern.error.limited.1", GTValues.VN[tier]))
+                        .addTooltips(limitedMsg)
                         .setExactLimit(1)
                         .or(abilities(MultiblockAbility.OUTPUT_ENERGY)).setExactLimit(1))
                 .where('H', states(getCasingState()).or(autoAbilities(false, true, false, false, true, true, true)))
```

Reply:
> This doesn't fix the turbines because the i18n call is executed in the parameter list (before we get to this function) in the large turbine's `createStructurePattern()`.. AFAICT there's no i18N usage here that needs to be saved though might as well not make the server execute it.
> 
> To fix the turbine, need to add something like this:
> 

Instead of explicitly checking the side, you could probably just pass `LocalizationUtils.format("translation key")` into the tooltip call